### ISSUE TITLE
Solidjs VList reactivity not working

### ISSUE THREAD

**Issue by imascha** [inokawa/virtua#425]
**Describe the bug**
As a minimal reproducable example, i created a `VList` with some example data and an Button that updates the example data.
But the DOM Elements don't update when the data is accessed trough the `VList` `item`, they only update when they are accessed trough the index. Is this wanted behaviour? 

Example: 
```
function App() {
  const [data, setData] = createSignal(["a", "b", "c", "d"]);
  return (
    <>
      <button onClick={() => setData(data().map((item) => item + item))}>
        Update data
      </button>
      <VList data={data()} style={{ height: "800px" }}>
        {(item, i) => (
          <div>
            {item} {data()[i]}
          </div>
        )}
      </VList>
    </>
  );
}
```

In this example only the `{data()[i]}` updates, when i click the button, the `{item}` never updates

**To Reproduce**
Here is a CodeSanbox of the Example: [link](https://codesandbox.io/p/devbox/nervous-sinoussi-p4z7hv?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clsw32mnx00072a66ukcznxwk%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clsw32mnx00022a666gburae6%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clsw32mnx00042a66j5h8okl0%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clsw32mnx00062a66yckplnur%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clsw32mnx00022a666gburae6%2522%253A%257B%2522id%2522%253A%2522clsw32mnx00022a666gburae6%2522%252C%2522tabs%2522%253A%255B%255D%257D%252C%2522clsw32mnx00062a66yckplnur%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clsw32mnx00052a66c2kpsq59%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clsw32mnx00062a66yckplnur%2522%252C%2522activeTabId%2522%253A%2522clsw32mnx00052a66c2kpsq59%2522%257D%252C%2522clsw32mnx00042a66j5h8okl0%2522%253A%257B%2522id%2522%253A%2522clsw32mnx00042a66j5h8okl0%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clsw32mnx00032a66vvz39vfj%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%252C%257B%2522id%2522%253A%2522cluikgkls00hx2a666cs9p8ol%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522cluikgl79000ddbe2gjk6fooj%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clsw32mnx00032a66vvz39vfj%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Expected behavior**
Both the  `{data()[i]}` and `{item}` should update, when i click the button

**Platform:**

- OS: Windows
- Browser: Chrome
- Version of this package: 0.29.1
- Version of framework: solid-js 1.8.14

**Comment by imascha** [comment]
I was expecting it to work similar to the `For` primative from solid. There the item is not a function. I don't know if that is going to be difficult to implement. I'll check out if it is something i can fix.

**Comment by inokawa** [comment]
Released in `0.29.2`. Thank you for the PR!
